From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: KurumiFake <kurumiisshidowife@gmail.com>
Date: Mon, 13 Jun 2022 05:53:08 +0200
Subject: [PATCH] rebrand to FloTinhHe


diff --git a/build.gradle.kts b/build.gradle.kts
index ee153a869bfa51d5a0b68a1af2352cd4fd5d8ebb..e8ef49bf163323fd3df88283995b85e15fda08dd 100644
--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -83,7 +83,7 @@ tasks.jar {
         attributes(
             "Main-Class" to "org.bukkit.craftbukkit.Main",
             "Implementation-Title" to "CraftBukkit",
-            "Implementation-Version" to "git-Starlivht-$implementationVersion", // Starlivht
+            "Implementation-Version" to "git-FloTinhHe-$implementationVersion", // Starlivht
             "Implementation-Vendor" to date, // Paper
             "Specification-Title" to "Bukkit",
             "Specification-Version" to project.version,
diff --git a/src/main/java/net/minecraft/server/MinecraftServer.java b/src/main/java/net/minecraft/server/MinecraftServer.java
index c0ea4f4c68ef8649842077a3d900452fd5417321..2be9cc2f62a0c9c7662559fb1c3a8649e9729c6a 100644
--- a/src/main/java/net/minecraft/server/MinecraftServer.java
+++ b/src/main/java/net/minecraft/server/MinecraftServer.java
@@ -1656,7 +1656,7 @@ public abstract class MinecraftServer extends ReentrantBlockableEventLoop<TickTa
 
     @DontObfuscate
     public String getServerModName() {
-        return "Starlivht"; // Starlivht > // Pufferfish - Pufferfish > // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
+        return "FloTinhHe"; // Starlivht > // Pufferfish - Pufferfish > // Paper - Paper > // Spigot - Spigot > // CraftBukkit - cb > vanilla!
     }
 
     public SystemReport fillSystemReport(SystemReport details) {
diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
index 6465ff5b7afe77d6a8b27c1caf4471ae37b4430e..1a855731f43e18b4234d6fda82e637513ef14af7 100644
--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
@@ -245,7 +245,7 @@ import javax.annotation.Nullable; // Paper
 import javax.annotation.Nonnull; // Paper
 
 public final class CraftServer implements Server {
-    private final String serverName = "Starlivht"; // Paper // Pufferfish // Starlivht
+    private final String serverName = "FloTinhHe"; // Paper // Pufferfish // Starlivht
     private final String serverVersion;
     private final String bukkitVersion = Versioning.getBukkitVersion();
     private final Logger logger = Logger.getLogger("Minecraft");
